source/MaterialXCore/Document.h

```diff
@@ -320,7 +320,7 @@ class Document : public GraphElement
                           const string& node = EMPTY_STRING)
     {
         NodeDefPtr child = addChild<NodeDef>(name);
-        child->setType(type);
+        child->addOutput("out", type);
```

Comment:
We no longer store a type but instead a child output called "out".

Reply:
Ah good point! That would definitely be an improvement as creating an output with either of those types doesn't make sense.